### PR TITLE
compile tests to target directory

### PR DESCRIPTION
**Pull request by drewbanin**
- compile schema+data test files to target/ dir
- wrap data tests in `select count(*) from ( .... )` so they align with schema tests

### CODE REVIEW

File: dbt/compilation.py [hunk 1]
```diff
@@ -380,10 +388,16 @@ def compile_nodes(self, linker, nodes, macro_generator):
                 injected_node['wrapped_sql'] = wrapped_stmt
                 wrapped_nodes[name] = injected_node
 
-            build_path = os.path.join('build', injected_node.get('path'))
+            if injected_node.get('resource_type') == NodeType.Test:
+                source_path = dbt.utils.get_pseduo_test_path(injected_node)
```
Comment (cmcarthur) [inline dbt/compilation.py]: let's move this to the parser later

File: dbt/compilation.py [hunk 2]
```diff
@@ -352,8 +352,16 @@ def compile_nodes(self, linker, nodes, macro_generator):
             # wrapping
             if injected_node.get('resource_type') in [NodeType.Test,
                                                       NodeType.Analysis]:
-                # don't wrap tests or analyses.
-                injected_node['wrapped_sql'] = injected_node['injected_sql']
+                if 'data' in injected_node['tags']:
+                    # data tests get wrapped in count(*)
+                    injected_node['wrapped_sql'] = (
+                        "select count(*) from (\n{test_sql}\n) sbq").format(
```
Comment (cmcarthur) [inline dbt/compilation.py]: this is ok for now, but i'm not crazy about inlining the sql here, and using a tag to decide whether or not to wrap. this means analyses with the data tag would get wrapped, which is a little odd.

File: dbt/utils.py [hunk 3]
```diff
@@ -195,3 +195,27 @@ def get_materialization(node):
...
+        test_type = 'schema_test'
+    elif 'data' in node.get('tags'):
+        test_type = 'data_test'
+    else:
+        test_type = 'test'
+
+    return test_type
+
+
+def get_pseduo_test_path(node):
```
Comment (cmcarthur) [inline dbt/utils.py]: pseudo

**Comment by drewbanin** [comment]
@cmcarthur made a couple of changes. That SQL is still in the compiler, but otherwise good feedback thanks :+1:

**Pull request merged**